src/utils/daos_hdlr.c

```diff
@@ -2566,95 +2576,241 @@ fs_copy(struct file_dfs *src_file_dfs,
 				"pool: %d\n", rc);
 			D_GOTO(out, rc);
 		}
-		rc = daos_cont_open(fa->src_poh, fa->src_c_uuid, DAOS_COO_RW,
-				    &fa->src_coh, src_cont_info, NULL);
+		rc = daos_cont_open(ca->src_poh, ca->src_c_uuid, DAOS_COO_RW,
+				    &ca->src_coh, src_cont_info, NULL);
 		if (rc != 0) {
 			fprintf(stderr, "failed to open source "
 				"container: %d\n", rc);
 			D_GOTO(err_src_root, rc);
 		}
-		rc = dfs_mount(fa->src_poh, fa->src_coh, O_RDWR,
-			       &src_file_dfs->dfs);
-		if (rc) {
-			fprintf(stderr, "dfs mount on source failed: %d\n", rc);
-			D_GOTO(err_src_dfs, rc);
+		if (is_posix_copy) {
+			rc = dfs_mount(ca->src_poh, ca->src_coh, O_RDWR,
+				       &src_file_dfs->dfs);
+			if (rc) {
+				fprintf(stderr, "dfs mount on source "
+					"failed: %d\n", rc);
+				D_GOTO(err_src, rc);
+			}
 		}
 	}
 
+	/* get source container type */
+	rc = dm_get_cont_prop(ca->src_coh, sysname, src_cont_info,
+			      ca->cont_prop_layout, &ca->cont_layout);
+	if (rc != 0) {
+		fprintf(stderr, "failed to query source "
+			"container: %d\n", rc);
+	}
+
+	/* get allocated oid on source container to set on destination */
+	rc = dm_get_cont_prop(ca->src_coh, sysname, src_cont_info,
+			      ca->cont_prop_oid, &ca->cont_oid);
+	if (rc != 0) {
+		fprintf(stderr, "failed to query source "
+			"container: %d\n", rc);
+	}
+
+	props = daos_prop_alloc(2);
+	if (props == NULL) {
+		fprintf(stderr, "Failed to allocate prop (%d)", rc);
+		D_GOTO(out, rc);
+	}
+	props->dpp_entries[0].dpe_type = ca->cont_prop_layout;
+	props->dpp_entries[0].dpe_val = ca->cont_layout;
+
+	/* only set max oid on created dest containers
+	 * if this is a non-posix copy
+	 */
+	if (!is_posix_copy) {
+		props->dpp_entries[1].dpe_type = ca->cont_prop_oid;
+		props->dpp_entries[1].dpe_val = ca->cont_oid;
+	}
+
 	/* open dst pool, dst cont, and mount dfs */
 	if (dst_file_dfs->type == DAOS) {
-		rc = daos_pool_connect(fa->dst_p_uuid, sysname,
-				       DAOS_PC_RW, &fa->dst_poh, NULL, NULL);
-		if (rc != 0) {
-			fprintf(stderr, "failed to connect to destination "
-				"pool: %d\n", rc);
-			D_GOTO(out, rc);
+		if (!uuid_is_null(ca->dst_p_uuid)) {
+			rc = daos_pool_connect(ca->dst_p_uuid, sysname,
+					       DAOS_PC_RW, &ca->dst_poh, NULL,
+					       NULL);
+			if (rc != 0) {
+				fprintf(stderr, "failed to connect to "
+					"destination pool: %d\n", rc);
+				D_GOTO(out, rc);
+			}
+		} else {
+			uuid_copy(ca->dst_p_uuid, ca->src_p_uuid);
+			rc = daos_pool_connect(ca->dst_p_uuid, sysname,
+					       DAOS_PC_RW, &ca->dst_poh,
+					       NULL, NULL);
+			if (rc != 0) {
+				fprintf(stderr, "failed to connect to "
+					"destination pool: %d\n", rc);
+				D_GOTO(err_src, rc);
+			}
+			uuid_copy(dattr.da_puuid, ca->dst_p_uuid);
+			uuid_copy(dattr.da_cuuid, ca->dst_c_uuid);
+			dattr.da_type = ca->cont_layout;
+			dattr.da_props = props;
+			rc = duns_create_path(ca->dst_poh, path,
+					      &dattr);
+			if (rc != 0) {
+				fprintf(stderr, "please provide a destination "
```

Comment:
style: im not sure we ever use "please" in our output.
maybe just print what the error is rather than a command to the user.